mlflow/R/mlflow/R/tracking-client.R

```diff
@@ -1,33 +1,89 @@
-new_mlflow_client <- function(tracking_uri, server_url = NULL) {
+
+new_mlflow_client <- function(tracking_uri) {
+  UseMethod("new_mlflow_client")
+}
+
+new_mlflow_uri <- function(raw_uri) {
+  parts <- strsplit(raw_uri, "://")[[1]]
+  structure(
+    list(scheme = parts[1], path = parts[2]),
+    class = c(paste("mlflow_", parts[1], sep = ""), "mlflow_uri")
+  )
+}
+
+new_mlflow_client_impl <- function(get_host_creds, get_cli_env = list, clazz = character()) {
```

Comment:
`class` isn't a reserved keyword in R and is a conventional argument for the subclass parameter in constructors